### PR TITLE
omniorb: Add TLS and on macOS CFNetwork support

### PR DESCRIPTION
**Pull request by tjuerges**
A couple of functional additions for omniORB:
- Enable TLS support by adding the `--enable-openssl` option.
- Enable on macOS CFNetwork support by adding the `--enable-cfnetwork` option.

And a couple of formula improvements:
- Change the parameters for `./configure` calls to use `std_configure_args `.
- Add the `--jobs` option to the `make` command.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/o/omniorb.rb [hunk 1]
```diff
@@ -36,15 +36,19 @@ def install
...
               /am_cv_python_version=`.*`/,
               "am_cv_python_version='#{xy}'"
-    system "./configure", "--prefix=#{prefix}"
-    system "make"
+    if OS.mac?
+      system "./configure", *std_configure_args, "--with-openssl", "--enable-cfnetwork"
+    else
+      system "./configure", *std_configure_args, "--with-openssl"
+    end
+    system "make", "--jobs=#{ENV.make_jobs}"
```
Comment (carlocab) [inline Formula/o/omniorb.rb:44]: `brew` already sets `MAKEFLAGS=-j#{ENV.make_jobs}`, so I don't think this is necessary.
```suggestion
    system "make"
```

File: Formula/o/omniorb.rb [hunk 2]
```diff
@@ -36,15 +36,19 @@ def install
     inreplace "configure",
               /am_cv_python_version=`.*`/,
               "am_cv_python_version='#{xy}'"
-    system "./configure", "--prefix=#{prefix}"
-    system "make"
+    if OS.mac?
+      system "./configure", *std_configure_args, "--with-openssl", "--enable-cfnetwork"
+    else
+      system "./configure", *std_configure_args, "--with-openssl"
+    end
```
Comment (carlocab) [inline Formula/o/omniorb.rb:43]:
```suggestion
    args = ["--with-openssl"]
    args << "--enable-cfnetwork"
    system "./configure", *args, *std_configure_args
```

**Review by carlocab — COMMENTED** [review]
Thanks! A few suggestions.

File: Formula/o/omniorb.rb
Comment (carlocab) [inline Formula/o/omniorb.rb:1]: Ah, looks like you need to update the dependencies to say
```ruby
depends_on "openssl@3"
uses_from_macos "zlib"
```

Comment (tjuerges) [reply Formula/o/omniorb.rb:1]: Will do that.

I suppose that I should squash all commits?

File: Formula/o/omniorb.rb [hunk 3]
```diff
@@ -36,15 +36,17 @@ def install
     inreplace "configure",
               /am_cv_python_version=`.*`/,
               "am_cv_python_version='#{xy}'"
-    system "./configure", "--prefix=#{prefix}"
+    args = ["--with-openssl"]
+    args << "--enable-cfnetwork"
```
Comment (carlocab) [inline Formula/o/omniorb.rb:40]:
```suggestion
    args << "--enable-cfnetwork" if OS.mac?
```

File: Formula/o/omniorb.rb
Comment (carlocab) [reply Formula/o/omniorb.rb:1]: Yes please!

**Review by carlocab — APPROVED** [review]
Thanks; great work!

**Pull request merged**